src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/finance/FinanceResourceIntTest.kt

```diff
@@ -205,4 +206,179 @@ class FinanceResourceIntTest : IntegrationTestBase() {
         .json("[]")
     }
   }
+
+  @Nested
+  @DisplayName("GET /transactions/{date}/first")
+  inner class GetIdAtDate {
+    @Nested
+    inner class Security {
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.get().uri("/transactions/2025-01-01/first")
+          .headers(setAuthorisation(roles = listOf()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.get().uri("/transactions/2025-01-01/first")
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access unauthorised with no auth token`() {
+        webTestClient.get().uri("/transactions/2025-01-01/first")
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+    }
+
+    @Test
+    fun getTransactionId() {
+      val id = webTestClient.get().uri("/transactions/2025-08-11/first")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_TRANSACTIONS")))
+        .exchange()
+        .expectStatus()
+        .isOk
+        .expectBody(Long::class.java)
+        .returnResult().responseBody!!
+      assertThat(id).isEqualTo(transaction2.transactionId)
+    }
+
+    @Test
+    fun `none found`() {
+      webTestClient.get().uri("/transactions/2025-08-12/first")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_TRANSACTIONS")))
+        .exchange()
+        .expectStatus()
+        .isNotFound
+        .expectBody()
+        .jsonPath("developerMessage").isEqualTo("No transactions found with date 2025-08-12")
+    }
+  }
+
+  @Nested
+  @DisplayName("GET /transactions/from/{transactionId}/all")
+  inner class NonOffenderTransactionsFromId {
+    @Nested
+    inner class Security {
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.get().uri("/transactions/from/99/1/1")
+          .headers(setAuthorisation(roles = listOf()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.get().uri("/transactions/from/99/1/1")
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access unauthorised with no auth token`() {
+        webTestClient.get().uri("/transactions/from/99/1/1")
+          .exchange()
+          .expectStatus().isUnauthorized
+      }
+    }
+
+    @Nested
+    inner class HappyPath {
+      @BeforeEach
+      fun setUp() {
+        nomisDataBuilder.build {
+          generalLedgerTransaction(10, 1, 1)
```

Comment:
it is worth adding parameter names here - especially as they are all the same type